mep/people/models.py

```diff
@@ -306,6 +307,14 @@ def save(self, *args, **kwargs):
 
         super(Person, self).save(*args, **kwargs)
 
+    def get_absolute_url(self):
+        '''
+        Return the public url to view library member's detail page
+        '''
+        # NOTE: using pk temporarily until we add slugs
+        # FIXME: should this return None for people without accounts?
```

Comment:
isn't `NoReverseMatch` expected behavior?

Reply:
It seems slightly different in my head, but I'm game for trying that. The main place I know it will come into play is the admin site, which I think will display a 'view on site' button based on the presence of the method on the model. But probably better to have it error than to have it generate a URL for something that isn't actually accessible... (we have a similar problem with CDH web profiles).

Reply:
I tested, seems to generate a server error no matter what somewhere in the admin view that tries to redirect. No reverse is a different error (on trying to resolve a url), so I'm just going to return `None` here for non-members (and test it).

Reply:
does that mean that having `{{ person.get_absolute_url }}` for a non-member anywhere in a template will throw a 500? would be nice if we could get it to at least render the page and have the link 404 instead...but i guess it's unlikely to occur anyway

Reply:
No, we'll probably get a `None` and a bad link if we try to do that - so the page will still render. 

The 500 is happening in the Django admin view that redirects to an object's absolute url. I don't see any way around it, but it's better than getting 404 reports (should we turn it on) for a URL that doesn't actually exist, as we're seeing right now for CDH web on occasion.